widget/src/scrollable.rs

```diff
@@ -1095,6 +1095,22 @@ impl Viewport {
         AbsoluteOffset { x, y }
     }
 
+    /// Returns the [`AbsoluteOffset`] of the current [`Viewport`], but with its
+    /// alignment reversed.
+    ///
+    /// This method can be useful to switch the alignment of a [`Scrollable`]
+    /// while maintaining its scrolling position.
+    pub fn absolute_offset_reversed(&self) -> AbsoluteOffset {
+        let AbsoluteOffset { x, y } = self.absolute_offset();
+
+        AbsoluteOffset {
+            x: ((self.content_bounds.width - self.bounds.width).max(0.0) - x)
+                .max(0.0),
```

Comment:
Nit: you shouldn't need both max since x will never be more than content - bounds.